### PR TITLE
168453305 - sea route admin tab - some routes missing

### PR DESCRIPTION
**Pull request by rckopteri**
# Fixed
* Admin sea route tab: Fixed sql query for routes which are missing calendar rules.

### CODE REVIEW

File: ote/src/cljs/ote/views/admin/sea_routes.cljs [hunk 1]
```diff
@@ -51,7 +51,8 @@
       3 ::transit/wednesday
       4 ::transit/thursday
       5 ::transit/friday
-      6 ::transit/saturday)))
+      6 ::transit/saturday
+      ::transit/sunday)))
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/admin/sea_routes.cljs]: Didn't go deep into this logic, so just double-checking is defaulting to sunday ok in all cases? It was already handled as index `0`?

File: ote/src/clj/ote/services/admin.sql [hunk 2]
```diff
@@ -121,7 +121,17 @@ SELECT
        r.modified AS "modified",
        r.created AS "created"
   FROM
-       "transit_route" r, unnest(r."service-calendars") c, unnest(c."service-rules") ru,
+       "transit_route" r,
+       lateral unnest(
```
Comment (philosotractor) [inline ote/src/clj/ote/services/admin.sql]:
```suggestion
       LATERAL unnest(
```

File: ote/src/clj/ote/services/admin.sql [hunk 3]
```diff
@@ -121,7 +121,17 @@ SELECT
...
+           CASE WHEN array_length(r."service-calendars", 1) >= 1
+                    THEN r."service-calendars"
+                ELSE '{null}'::transit_service_calendar[]
+           END) c,
+       lateral unnest (
```
Comment (philosotractor) [inline ote/src/clj/ote/services/admin.sql]:
```suggestion
       LATERAL unnest (
```

File: ote/src/cljs/ote/views/admin/sea_routes.cljs (hunk 1 above)
Comment (rckopteri) [reply ote/src/cljs/ote/views/admin/sea_routes.cljs]: Yes, it is ok.

If service calendar and service rule is not given we don't have any dates when route is supposed to be active (ajopäivät, don't know what it is in english). This prevents errors when ajopäivät is not given.

**Pull request merged**